core/src/main/scala/sttp/client3/SttpBackendOptions.scala

```diff
@@ -59,9 +60,14 @@ object SttpBackendOptions {
       }
     }
 
-    def ignoreProxy(host: String): Boolean = {
+    def ignoreProxy(host: String): Boolean =
+      matchesNonProxyHost(host) || doesNotMatchAnyHostToProxy(host)
+
+    private def matchesNonProxyHost(host: String) =
       nonProxyHosts.exists(isWildCardMatch(host, _))
-    }
+
+    private def doesNotMatchAnyHostToProxy(host: String) =
+      hostsToProxy != Nil && hostsToProxy.forall(!isWildCardMatch(host, _))
```

Comment:
The `hostsToProxy != Nil` is necessary because otherwise we'd be ignoring proxy despite user not setting any `hostsToProxy`. The reason being that `forall` returns `true` by default for `Nil`, which makes sense of course, but in this case we don't want to check this condition at all if user did not set `hostsToProxy`.

Now that I've written it down, I think that wrapping `hostsToProxy` in `Option` could be best here, but that would be kinda different from `nonProxyHosts`... What do you think?

Reply:
I've rewritten it as `hostsToProxy != Nil && !hostsToProxy.exists(isWildCardMatch(host, _))`, because I think it might be clearer that way.